index.js

```diff
@@ -11,8 +11,16 @@ function copyBuffer (cur) {
 
 function rfdc (opts) {
   opts = opts || {}
-
   if (opts.circles) return rfdcCircles(opts)
+
+  const constructorHandlers = new Map([
+    [Date, (o) => new Date(o)],
+    [Map, (o, fn) => new Map(cloneArray(Array.from(o), fn))],
+    [Set, (o, fn) => new Set(cloneArray(Array.from(o), fn))]
+  ].concat(opts.constructorHandlers || []))
```

Comment:
```suggestion
  const constructorHandlers = new Map()
  constructorHandlers.add(Date, (o) => new Date(o))
  constructorHandlers.add(Map, (o, fn) => new Map(cloneArray(Array.from(o), fn)))
  constructorHandlers.add(Set, (o, fn) => new Set(cloneArray(Array.from(o), fn)))

  if (opts.constructorHandlers) {
    opts.constructorHandlers.forEach((name, handler) => constructorHandlers.add(name, handler)
  }
```

It is faster to work around the additional array allocation. That could also be done inside of the Map and Set handlers itself.

Reply:
I don't see much difference in the benchmark, but will make the change anyway

```
benchRfdc*100: 206.686ms
benchRfdcProto*100: 218.722ms
benchRfdcCircles*100: 228.578ms
benchRfdcCirclesProto*100: 239.244ms
```